ddl/ddl_api.go

```diff
@@ -4479,7 +4449,7 @@ func (d *ddl) getModifiableColumnJob(ctx context.Context, sctx sessionctx.Contex
 			Location:      &model.TimeZoneLocation{Name: tzName, Offset: tzOffset},
 		},
 		CtxVars: []interface{}{needChangeColData},
-		Args:    []interface{}{&newCol, originalColName, spec.Position, modifyColumnTp, newAutoRandBits},
+		Args:    []interface{}{&newCol.ColumnInfo, originalColName, spec.Position, modifyColumnTp, newAutoRandBits},
```

Comment:
Why change this arg's type?

Reply:
Using `newCol` can bring some unnecessary type conversions.

It is compatible because the decoding part use `*model.ColumnInfo` to receive it.